esmvaltool/cmorizers/data/formatters/datasets/esacci_cloud.py

```diff
@@ -0,0 +1,404 @@
+"""ESMValTool CMORizer for ESACCI-CLOUD data.
+
+Tier
+    Tier 2: other freely-available dataset.
+
+Source
+    https://public.satproj.klima.dwd.de/data/ESA_Cloud_CCI/CLD_PRODUCTS/v3.0/L3U/AVHRR-PM/
+
+Last access
+    20230619
+
+Download and processing instructions
+    see downloading script
+"""
+
+import copy
+import glob
+import logging
+import os
+from calendar import monthrange
+from datetime import datetime
+
+import cf_units
+import iris
+from dask import array as da
+from esmvalcore.preprocessor import (
+    daily_statistics,
+    monthly_statistics,
+    regrid,
+)
+from iris import NameConstraint
+
+from esmvaltool.cmorizers.data import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+
+def _create_nan_cube(cube, year, month, day):
+    """Create cube containing only nan from existing cube."""
+    nan_cube = cube.copy()
+    nan_cube.data = da.ma.masked_greater(cube.core_data(), -1e20)
+
+    # Read dataset time unit and calendar from file
+    dataset_time_unit = str(nan_cube.coord("time").units)
+    dataset_time_calender = nan_cube.coord("time").units.calendar
+    newtime = datetime(
+        year=year,
+        month=month,
+        day=day,
+        hour=12,
+        minute=0,
+        second=0,
+        microsecond=0,
+    )
+    newtime_num = cf_units.date2num(
+        newtime, dataset_time_unit, dataset_time_calender
+    )
+    nan_cube.coord("time").points = float(newtime_num)
+
+    return nan_cube
+
+
+def _handle_missing_day(year, month, iday, short_name, cubes, cubes_day):
+    """Fill missing day."""
+    if short_name in ["clt", "ctp"]:
+        daily_cube = _create_nan_cube(cubes_day[0], year, month, iday)
+        cubes.append(daily_cube)
+    daily_cube_day = _create_nan_cube(cubes_day[0], year, month, iday)
+    cubes_day.append(daily_cube_day)
```

Comment:
Why do you need to produce two cube lists (`cubes` and `cubes_day`) for the variables clt and ctp? What's the difference between `cubes` and `cubes_day`?